src/find/matchers/exec.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2017 Google Inc.
+//
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+use std::error::Error;
+use std::ffi::OsString;
+use std::io::{Write, stderr};
+use std::path::Path;
+use std::process::Command;
+use walkdir::DirEntry;
+
+use find::matchers::{Matcher, MatcherIO};
+
+pub struct SingleExecMatcher {
+    executable: String,
+    args: Vec<Option<OsString>>,
```

Comment:
So this is very much a style point, but Option doesn't feel like the right type for this; {} is not the absence of an arg, but the presence of the filename placeholder. I'd personally do `enum Arg { Filename, LiteralArg(OsString) }` and `args: Vec<Arg`>.